napari/plugins/_npe2.py

```diff
@@ -501,18 +514,44 @@ def _register_manifest_actions(mf: PluginManifest) -> None:
     app = get_app()
     actions, submenus = _npe2_manifest_to_actions(mf)
     samples_submenu, sample_actions = _get_samples_submenu_actions(mf)
+    widgets_submenu: List[Any] = []
+    widget_actions: List[Any] = []
+    try:
+        from napari._qt._qplugins import _get_widgets_submenu_actions
+    except ModuleNotFoundError:
+        pass
+    else:
+        widgets_submenu, widget_actions = _get_widgets_submenu_actions(mf)
+
     context = pm.get_context(cast('PluginName', mf.name))
-    # Connect 'unregister' callback to plugin deactivate ('unregistered') event
+
+    # Register and connect dispose callback to plugin deactivate ('unregistered') event
+    actions = actions + sample_actions + widget_actions
     if actions:
         context.register_disposable(app.register_actions(actions))
+    submenus = submenus + samples_submenu + widgets_submenu
     if submenus:
         context.register_disposable(app.menus.append_menu_items(submenus))
-    if samples_submenu:
-        context.register_disposable(
-            app.menus.append_menu_items(samples_submenu)
-        )
-    if sample_actions:
-        context.register_disposable(app.register_actions(sample_actions))
+
+    # Register dispose functions that remove plugin widgets from widget dictionary
+    # `window._dock_widgets` but ONLY if Qt present.
+    try:
+        from napari._qt._qapp_model.qactions._qproviders import _provide_window
+    except ModuleNotFoundError:
+        pass
+    else:
+        if window := _provide_window():
+
+            class Event(NamedTuple):
+                value: str
```

Comment:
Is there a reason not to make this an actual `napari.utils.events.Event? Just that this is lighter weight?

Reply:
Lightweight. `_remove_dock_widget` does not need more than something with a `.value` and seems a waste?

Reply:
Thanks - that makes sense. I don't feel strongly about this, but the local class creation caught my eye.

I could see this being confusing if debugging, or if someone ends up modifying `_remove_dock_widget`.

One alternative idea (take or leave): move this small class def to the top-level scope and add a docstring.

Reply:
Happy to amend, @DragaDoncila WDYT?

Reply:
explicit is better than implicit, probably best to create a real `Event`, yeah